contracts/TokenDistributor.sol

```diff
@@ -75,24 +75,31 @@ contract TokenDistributor is ITokenDistributor {
         uint previousCheckpoint = lastCheckpoint;
         uint sinceLast = block.timestamp - previousCheckpoint;
         lastCheckpoint = block.timestamp;
+        /// @dev this/nextWeek is for iterating through time
         uint thisWeek = _startOfWeek(previousCheckpoint);
         uint nextWeek = 0;
 
+        /// @dev this/nextEpoch is for mapping tokens to a certain
+        /// epoch in tokensPerEpoch
+        uint thisEpoch = (_startOfWeek(previousCheckpoint) - startTime) / 1 weeks;
+        uint nextEpoch = 0;
+
         /// @dev Loop for potential missed weeks
         /// iterates until caught up, unlikely to go to 52 weeks
         for (uint i = 0; i < WEEKS_IN_YEAR; i++) {
             nextWeek = thisWeek + 1 weeks;
+            nextEpoch = thisEpoch + 1;
 
             if (block.timestamp < nextWeek) {
                 /// @dev if in the current week
                 if (sinceLast == 0) {
                     /// @dev If no time change since last checkpoint just add new tokens
                     /// that may have been deposited (same block)
-                    tokensPerEpoch[thisWeek] += toDistribute;
+                    tokensPerEpoch[thisEpoch] += toDistribute;
                 } else {
                     /// @dev In the event that toDistribute contains tokens
                     /// for multiple weeks we take the remaining portion
-                    tokensPerEpoch[thisWeek] +=
+                    tokensPerEpoch[thisEpoch] +=
```

Comment:
Instead of creating a whole new `nextEpoch` variable you need to keep track of (risking errors) I would just do the conversion right before you store the value for example
```solidity
uint epoch = (thisWeek - startTime) / 1 weeks;
tokensPerEpoch[epoch] += toDistribute;
```

also break it out into a function called `epochFromTimestamp(uint timestamp)`